modules/components/analytics/core/org.wso2.carbon.pc.analytics.core/src/main/java/org/wso2/carbon/pc/analytics/core/utils/Helper.java

```diff
@@ -0,0 +1,339 @@
+	 * @return a sorted list as a JSON array string
+	 * @throws JSONException
+	 */
+	public static String getDoubleValueSortedList(Hashtable<String, Double> table, String key1, String key2, String order, int count)
+																			throws JSONException {
+		//Transfer as List and sort it
+		ArrayList<Map.Entry<String, Double>> l = new ArrayList(table.entrySet());
+		Collections.sort(l, new Comparator<Map.Entry<String, Double>>() {
+			public int compare(Map.Entry<String, Double> o1, Map.Entry<String, Double> o2) {
+				return o1.getValue().compareTo(o2.getValue());
+			}
+		});
+		JSONArray array = new JSONArray();
+		for (int i = 0 ; i < l.size() ; i++){
+			JSONObject o = new JSONObject();
+			o.put(key1, l.get(i).getKey());
+			o.put(key2, round(l.get(i).getValue(), 2));
+			array.put(o);
+		}
+
+		//if count exceeds the array length, then assign the array length to the count variable
+		if(count > array.length()){
+			count = array.length();
+		}
+
+		JSONArray arrayPortion = new JSONArray();
+		if(order.equalsIgnoreCase(AnalyticConstants.TOP)){
+			for (int i = array.length() - count ; i < array.length() ; i++){
+				arrayPortion.put(array.get(i));
+			}
+		}else if(order.equalsIgnoreCase(AnalyticConstants.BOTTOM)){
+			for (int i = 0 ; i < count ; i++){
+				arrayPortion.put(array.get(i));
+			}
+		}
+		return arrayPortion.toString();
+	}
+
+	/**
+	 * Get sorted list (sort by int type values)
+	 *
+	 * @param table is a hash table to keep the result as key-value pairs
+	 * @param key1 is the name for the first value of the JSON object
+	 * @param key2 is the name for the second value for the JSON object
+	 * @param order is to get the top or bottom results
+	 * @param count is to limit the number of results
+	 * @return a sorted list as a JSON array string
+	 * @throws JSONException
+	 */
+	public static String getIntegerValueSortedList(Hashtable<String, Integer> table, String key1, String key2, String order, int count)
+																			throws JSONException {
+		ArrayList<Map.Entry<String, Integer>> l = new ArrayList(table.entrySet());
+		Collections.sort(l, new Comparator<Map.Entry<String, Integer>>() {
+			public int compare(Map.Entry<String, Integer> o1, Map.Entry<String, Integer> o2) {
+				return o1.getValue().compareTo(o2.getValue());
+			}
+		});
+		JSONArray array = new JSONArray();
+		for (int i = 0 ; i < l.size() ; i++){
+			JSONObject o = new JSONObject();
+			o.put(key1, l.get(i).getKey());
+			o.put(key2, l.get(i).getValue());
+			array.put(o);
+		}
+
+		//if count exceeds the array length, then assign the array length to the count variable
+		if(count > array.length()){
+			count = array.length();
+		}
+
+		JSONArray arrayPortion = new JSONArray();
+		if(order.equalsIgnoreCase(AnalyticConstants.TOP)){
+			for (int i = array.length() - count ; i < array.length() ; i++){
+				arrayPortion.put(array.get(i));
+			}
+		}else if(order.equalsIgnoreCase(AnalyticConstants.BOTTOM)){
```

Comment:
Fxed in https://github.com/wso2/product-pc/pull/36. (This class renamed as "AnalyticsUtils")